### PR TITLE
Ignore dev component versions for update pull-request

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Ignores dev versions for update pull requests.
Release versions are identified by the string `-dev-`.

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Comment by ccwienk** [comment]
I would prefer to make this configurable via pipeline-definition

File: version.py [hunk 1]
```diff
@@ -211,11 +211,13 @@ def process_version(
     return processed_version
 
 
-def find_latest_version(versions) -> str:
+def find_latest_version(versions, ignore_dev_versions=False) -> str:
```
Comment (ccwienk) [inline version.py:214]: I think we should rather parse to semver (there is a util function for that in version.py), and check the parsing result for being "final"

**Review by ccwienk — CHANGES_REQUESTED** [review]

File: version.py (hunk 1 above)
Comment (schrodit) [reply version.py:214]: but we should still have the option to configure if dev versions should be ignored?

Comment (schrodit) [reply version.py:214]: should it be also possible to ignore dev versions in the other helper function like `find_latest_version_with_matching_minor` ?

**Comment by AndreasBurger** [comment]
Closed in favor of #581 (contains everything here plus some small changes from me)

**Pull request closed without merging**